libopenage/renderer/stages/terrain/terrain_mesh.cpp

```diff
@@ -97,10 +97,11 @@ const std::shared_ptr<renderer::UniformInput> &TerrainRenderMesh::get_uniforms()
 	return this->uniforms;
 }
 
-Eigen::Matrix4f TerrainRenderMesh::get_model_matrix() {
+void TerrainRenderMesh::create_model_matrix(const coord::scene2_delta &offset) {
```

Comment:
when is this called? since it updates it repeatedly, should be called `update_model_matrix`? and that's kinda the other way round, it could be better it the terrainmesh generates its new model matrix by querying somewhere else + caching of an already-current result?

Reply:
It's only called when the mesh is created or changed (in `fetch_updates()`). This doesn't happen every frame though and it will probably happen only once during a game in the final implementation.

> it could be better it the terrainmesh generates its new model matrix by querying somewhere else + caching of an already-current result?

this is basically what it already does. The created model matrix is cached and reused in `update_uniforms`.